libraries/chain/include/scorum/chain/services/budgets.hpp

```diff
@@ -132,6 +162,22 @@ class dbs_advertising_budget : public dbs_service_base<typename budget_service_t
     void finish_budget(const oid<adv_budget_object<budget_type_v>>& id) override;
     void close_empty_budgets() override;
 
+    template <typename C> void update_totals(C&& callback)
```

Comment:
Make this method private. Also it is better I guess not to use template here. It is better to used std::function<> here for now. We could update this later using function_view for example.